src/request.ts

```diff
@@ -35,6 +35,16 @@ const request: {
 	});
 });
 
+[ 'POST', 'PUT' ].forEach(method => {
+	Object.defineProperty(request, method.toLowerCase(), {
+		value(url: string, options: RequestOptions = {}): UploadObservableTask<Response> {
+			options = Object.create(options);
+			options.method = method;
+			return <UploadObservableTask<Response>> request(url, options);
```

Comment:
While it isn't a big deal, we are trying to move to `return request(url, options) as UploadObservableTask<Response>;`  The TypeScript team is recommending it to avoid issues with JSX.  Though we don't plan to use JSX, it still feels cleaner/more readable.